src/core/channel/http_client_filter.c

```diff
@@ -43,8 +43,9 @@ typedef struct call_data {
 
   int got_initial_metadata;
   grpc_stream_op_buffer *recv_ops;
-  void (*on_done_recv)(void *user_data, int success);
-  void *recv_user_data;
+  grpc_iomgr_closure *on_done_recv;
+
+  grpc_iomgr_closure hc_on_recv;
```

Comment:
Can we get a comment describing the new field hc_on_recv (or for that matter, the other fields too!). What distinguishes this from on_done_recv?